### PR TITLE
fix(driver): fix memfd detection in the kmod 

### PR DESCRIPTION
**Pull request by Andreagit97**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area driver-kmod

/area tests

**Does this PR require a change in the driver versions?**

Already bumped

**What this PR does / why we need it**:

This PR adds some test to memfd detection and fix a bug in the kmod (tests -> :heart: )

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by FedeDP — APPROVED** [review]
Great fix, as always @Andreagit97 !
/approve

File: driver/modern_bpf/helpers/extract/extract_from_kernel.h [hunk 1]
```diff
@@ -769,7 +769,6 @@ static __always_inline bool extract__exe_from_memfd(struct file *file)
 		return false;
 	}
 
-#pragma unroll
```
Comment (incertum) [inline driver/modern_bpf/helpers/extract/extract_from_kernel.h:772]: @Andreagit97 why remove it? Seems unrelated to the kmod fix and why only do in modern_bpf, it's typically used btw.

Comment (Andreagit97) [reply driver/modern_bpf/helpers/extract/extract_from_kernel.h:772]: in the modern bpf, we don't use the unroll, it is something we have to use in the old one because bounded loops were introduced in kernel 5.3 :/ IIRC there are no other places in the modern probe in which we use the unroll :thinking:

**Review by incertum — APPROVED** [review]
/approve
